eng/common/pipelines/templates/validate-filename.yml

```
@@ -0,0 +1,16 @@
+parameters:
+  WorkingDirectory: $(System.DefaultWorkingDirectory)
+steps:
+- pwsh: |
+    $differByCaseFiles = git ls-files | Group-Object | Where-Object { $_.Count -gt 1 } | ForEach-Object { $_.Group} 
+
+    if ($differByCaseFiles) 
+    {
+      Write-Error "Do NOT name the files which only differ in case. Please check and rename the following files:"
+      $differByCaseFiles | Write-Error
+      exit 1
+    }
+    Write-Host "There are no file names that only differ in case."
+  displayName: Validate file names do not only differ on case
```

This isn't going to display well in the UI given the length. Perhaps something like this?

```suggestion
Check file case duplicates
```